risc0/r0vm/src/lib.rs

```diff
@@ -174,14 +174,13 @@ impl Cli {
             HashFn::Sha256 => "sha-256",
             HashFn::Poseidon2 => "poseidon2",
         };
-        let opts = ProverOpts {
-            hashfn: hashfn.to_string(),
-            prove_guest_errors: self.prove_guest_errors,
-            receipt_kind: match self.receipt_kind {
-                ReceiptKind::Composite => risc0_zkvm::ReceiptKind::Composite,
-                ReceiptKind::Succinct => risc0_zkvm::ReceiptKind::Succinct,
-                ReceiptKind::Compact => risc0_zkvm::ReceiptKind::Compact,
-            },
+        let mut opts = ProverOpts::default();
+        opts.hashfn = hashfn.to_string();
+        opts.prove_guest_errors = self.prove_guest_errors;
+        opts.receipt_kind = match self.receipt_kind {
+            ReceiptKind::Composite => risc0_zkvm::ReceiptKind::Composite,
+            ReceiptKind::Succinct => risc0_zkvm::ReceiptKind::Succinct,
+            ReceiptKind::Compact => risc0_zkvm::ReceiptKind::Compact,
```

Comment:
I prefer to avoid using `mut` as much as possible to prevent future bugs.

Reply:
In 4fdda5fb03057db0228d3d9d367307c793f9732c I addressed this by adding a kind of builder pattern to the `ProverOpts`. Note that the reason for this change is that I added `#[non_exhaustive]` to `ProverOpts`. Without `#[non_exhaustive]`, because all the fields are public, it is a breaking change to add new fields.
https://github.com/risc0/risc0/blob/4fdda5fb03057db0228d3d9d367307c793f9732c/risc0/zkvm/src/host/client/prove/mod.rs#L122-L123